server/neptune/sync/exec.go

```diff
@@ -0,0 +1,70 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/logger"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+// RunNewProcessGroupCommand is useful for running separate commands that shouldn't receive termination
+// signals at the same time as the parent process
+func RunNewProcessGroupCommand(ctx context.Context, cmd *exec.Cmd, cmdName string) error {
+	// should this be defined by struct creating the cmd or this helper fxn?
+	// could also replace this with a check confirming this is set and erroring if not
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setpgid: true,
+	}
+	if err := cmd.Start(); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("starting %s command", cmdName))
+	}
+	done := make(chan struct{})
+	defer close(done)
+	go terminateProcessOnCtxCancellation(ctx, cmd.Process, done)
+
+	err := cmd.Wait()
+	if ctx.Err() != nil {
+		return errors.Wrap(ctx.Err(), fmt.Sprintf("waiting for %s process", cmdName))
+	}
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("waiting for %s process", cmdName))
+	}
+	return nil
+}
+
+func terminateProcessOnCtxCancellation(ctx context.Context, p *os.Process, processDone chan struct{}) {
```

Comment:
Does this need to be in its own function?

Reply:
nope, it was just a personal preference on making it easier to read, am willing to change but just gonna wait for the initial pr to merge first: https://github.com/lyft/atlantis/pull/537